contracts/exchange/types/DataStructure.sol

```diff
@@ -54,6 +54,7 @@ function currencyIsValid(Currency iter) pure returns (bool) {
   return iter > type(Currency).min && iter <= type(Currency).max;
 }
 
+uint64 constant MAX_BALANCE_DECIMALS = 9;
```

Comment:
I suppose this is something that is subject to change in the future (eg if we onboard DOGE). But it seems migratable